java/src/jmri/jmrix/mqtt/MqttAdapter.java

```diff
@@ -161,8 +161,9 @@ public void connect() throws IOException {
             }
 
             // generate a unique client ID based on the network ID and the system prefix of the MQTT connection.
-            String clientID = jmri.util.node.NodeIdentity.networkIdentity() + getSystemPrefix();
-
+            String clientID = jmri.InstanceManager.getDefault(jmri.web.server.WebServerPreferences.class).getRailroadName() + " "
+                                + getSystemPrefix() + " " + jmri.util.node.NodeIdentity.networkIdentity();
+            
             // ensure that only valid characters are included in the client ID
             clientID = clientID.replaceAll("[^A-Za-z0-9]", "");
```

Comment:
I'm reading this on a mobile so I'm not fully sure I read this correctly. But it seems to me that all the spaces in the client is will be removed.

Is it true that only letters and digits are allowed in the client id? Often underscore is allowed as well, and if that's the case, I would have replaced spaces with underscore.

Reply:
You're right:  Only upper/lower case letters and digits are allowed.  No spaces, no punctuation.  I've removed the redundant spaces.  Thanks for the careful reading!